src/main/java/dk/alexandra/fresco/lib/helper/builder/AbstractProtocolBuilder.java

```diff
@@ -146,7 +177,19 @@ public ProtocolProducer getProtocol() {
 	
 	@Override
 	public void reset() {
+		if(parent != null){
+			parent.reset();
+			return;
+		}
 		curpp = new SequentialProtocolProducer();
 		producerStack = new Stack<AppendableProtocolProducer>();
 	}
+	
+	/**
+	 * Sets the parent of this builder meaning that the internal stack of the builder will be the parents. 
+	 * @param parent
+	 */
+	protected void setParentBuilder(AbstractProtocolBuilder parent){
```

Comment:
Such that a functional builder needs a master to function? 

Sure, we could investigate that - then the question would be which builders are structural. The Omnibuilder for sure, but other than that right now, I don't know.